daemon/analytics.c

```diff
@@ -1,5 +1,41 @@
 #include "common.h"
 
+void analytics_main_cleanup(void *ptr) {
+    struct netdata_static_thread *static_thread = (struct netdata_static_thread *)ptr;
+    static_thread->enabled = NETDATA_MAIN_THREAD_EXITING;
+
+    debug(D_ANALYTICS, "Cleaning up...");
+
```

Comment:
Another suggestion, I think you can clean here the thread allocated inside `daemon/main.c` and also the variable that you used in another function.